lib/mappings/overview.js

```diff
@@ -0,0 +1,77 @@
+import { getWallets, getWalletInfo } from '../actions';
+import assert from 'bsert';
+
+import {
+  selectWallets,
+  selectMultisigWallets,
+  selectWalletInfo,
+  selectMultisigWalletInfo,
+} from './selectors';
+
+// TODO: append proper network symbol to balance based on chain
+function buildWalletsList(wallets, multisigWallets) {
```

Comment:
So this is basically the ideal use case of `reselect` and selectors since you're going through the stores and re-shaping the data based on a specific use case. I think what will likely happen in this implementation is that every time `walletsList` is retrieved through the `mapStateToProps` below is that this function will be called, which will go through the for loops. If you use `reselect` (which is available as a peerDependency too for plugins), then the walletsList will be memoized and returned, only recalculated if the state gets updated.